### PR TITLE
Explain that Go does not include occurrences in check definitions or built-in filters

### LINKED ISSUE #1793: Need to clarify occurrences differences in Classic v Go
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

## Expected Behavior

@rgeniesse has noticed that there have been several occurrences of folks in the community thinking that `occurrences` is part of the check definition like it was in classic. This is no longer the case. We should make a note in the migration guide about this, and possible one in the check/events reference material

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds note to explain Go does not include occurrences filter in Migration guide and Filters reference
- Adds note to explain Go check definitions do not include occurrences attribute in Check reference

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/1793

### CODE REVIEW

**Comment by hillaryfraley** [comment]
I think this finally includes everything requested in https://github.com/sensu/sensu-docs/issues/1793 and is ready for review.

File: content/sensu-go/5.19/installation/upgrade.md [hunk 1]
```diff
@@ -144,12 +144,23 @@ You can use the built-in [incidents filter][9] to recreate the Sensu Core 1.x be
...
+
+#### Fatigue check filter asset
+
+In Sensu Go, the [fatigue check filter asset][11] uses annotations to tune the number of occurrences and interval for alerts on a per-check or per-entity basis and control whether resolution events will pass through.
+
+The [Sensu Translator version 1.1.0][18] retrieves occurrence and refresh values from a Sensu Core 1.x check definition and outputs them as annotation values in a Sensu Go check definition.
+The [fatigue check filter asset][11] uses these [annotations][33] to manage alert fatigue.
+
+However, the Sensu Translator doesn't automatically add the fatigue check filter asset or the filter configuration you need to run it.
+To use the fatigue check filter asset, you must [register it][15] and create a correctly configured [event filter definition][19].
```
Comment (cwjohnston) [inline content/sensu-go/5.19/installation/upgrade.md]: I think there's a third step here: add the event filter to the list of filters on applicable handlers

File: content/sensu-go/5.19/reference/checks.md [hunk 2]
```diff
@@ -479,6 +479,10 @@ Learn how to use check hooks with the [Sensu hooks reference documentation][6].
 
 ## Check specification
 
+{{% notice note %}}
+**NOTE**: The Sensu Go check definition does not include the `occurrences` attribute. In Sensu Go, [occurrences are included in events][60].
```
Comment (cwjohnston) [inline content/sensu-go/5.19/reference/checks.md]: Occurrences are also tracked in Sensu Classic events, so the distinction implied here isn't very meaningful to users.

Classic had no concept of metadata, instead arbitrary key/value pairs could be added directly along side other attributes which now go in the `spec` block. `occurrences` in the check definition started out as arbitrary but became a convention.

**Review by cwjohnston — CHANGES_REQUESTED** [review]
A couple comments here. I think overall this is on the right track.

File: content/sensu-go/5.19/installation/upgrade.md (hunk 1 above)
Comment (hillaryfraley) [inline content/sensu-go/5.19/installation/upgrade.md]:
```suggestion
To use the fatigue check filter asset, you must [register it][15], create a correctly configured [event filter definition][19], and [add the event filter][34] to the list of filters on applicable handlers.
```

File: content/sensu-go/5.19/installation/upgrade.md [hunk 3]
```diff
@@ -209,3 +223,4 @@ See the metadata attributes section in the reference documentation for more info
 [30]: /images/web-ui-entity-warning.png
 [31]: https://sensu.io/contact?subject=contact-sales/
 [32]: https://blog.sensu.io/one-year-of-sensu-go
+[33]: https://github.com/nixwiz/sensu-go-fatigue-check-filter/#configuration-1
```
Comment (hillaryfraley) [inline content/sensu-go/5.19/installation/upgrade.md:226]:
```suggestion
[33]: https://github.com/nixwiz/sensu-go-fatigue-check-filter/#configuration-1
[34]: ../../guides/reduce-alert-fatigue/#assign-the-event-filter-to-a-handler-1
```

File: content/sensu-go/5.19/reference/checks.md (hunk 2 above)
Comment (hillaryfraley) [inline content/sensu-go/5.19/reference/checks.md]:
```suggestion
**NOTE**: In Sensu Go, the `occurrences` attribute is not part of the check definition like it was in Sensu Core.
```

Comment (hillaryfraley) [reply content/sensu-go/5.19/reference/checks.md]: Thank you -- I think I phrased that poorly. Does my update make more sense?

File: content/sensu-go/5.19/installation/upgrade.md [hunk 4]
```diff
@@ -144,12 +144,23 @@ You can use the built-in [incidents filter][9] to recreate the Sensu Core 1.x be
...
-As a result, the built-in occurrences event filter in Sensu Core 1.x is not included in Sensu Go, but you can replicate its functionality with [the repeated events filter definition][10].
-
 Sensu Go includes three new built-in [event filters][9]: only-incidents, only-metrics, and allow-silencing.
 Sensu Go does not include a built-in check dependencies filter or a filter-when feature.
 
+Ruby eval logic from Sensu Core 1.x is replaced with JavaScript expressions in Sensu Go, opening up powerful ways to filter events based on occurrences and other event attributes.
+As a result, **Sensu Go does not include the built-in occurrence-based event filter in Sensu Core 1.x**, which allowed you to control the number of duplicate events that reached the handler.
+You can replicate the occurrence-based filter's functionality with Sensu Go's [repeated events filter definition][10].
+
+#### Fatigue check filter asset
```
Comment (cwjohnston) [inline content/sensu-go/5.19/installation/upgrade.md]:
```suggestion
#### Fatigue check filter
```

File: content/sensu-go/5.19/installation/upgrade.md [hunk 5]
```diff
@@ -144,12 +144,23 @@ You can use the built-in [incidents filter][9] to recreate the Sensu Core 1.x be
...
+
+In Sensu Go, the [fatigue check filter asset][11] uses annotations to tune the number of occurrences and interval for alerts on a per-check or per-entity basis and control whether resolution events will pass through.
```
Comment (cwjohnston) [inline content/sensu-go/5.19/installation/upgrade.md]:
```suggestion
For Sensu Go users, we recommend the [fatigue check filter][11], a JavaScript implementation of the `occurrences` filter from Sensu 1.x.
This filter looks for [check and entity annotations][33] in each event it receives and uses the values of those annotations to configure the filter's behavior on a per-event basis.
```

File: content/sensu-go/5.19/installation/upgrade.md [hunk 6]
```diff
@@ -144,12 +144,23 @@ You can use the built-in [incidents filter][9] to recreate the Sensu Core 1.x be
...
+
+The [Sensu Translator version 1.1.0][18] retrieves occurrence and refresh values from a Sensu Core 1.x check definition and outputs them as annotation values in a Sensu Go check definition.
```
Comment (cwjohnston) [inline content/sensu-go/5.19/installation/upgrade.md]:
```suggestion
The [Sensu Translator version 1.1.0][18] retrieves occurrence and refresh values from a Sensu Core 1.x check definition and outputs them as annotations in a Sensu Go check definition, compatible with the fatigue check filter.
```

File: content/sensu-go/5.19/installation/upgrade.md [hunk 7]
```diff
@@ -144,12 +144,23 @@ You can use the built-in [incidents filter][9] to recreate the Sensu Core 1.x be
...
+The [fatigue check filter asset][11] uses these [annotations][33] to manage alert fatigue.
```
Comment (cwjohnston) [inline content/sensu-go/5.19/installation/upgrade.md]:
```suggestion
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

**Pull request merged**